state/immutable-trie/snapshot.go

```diff
@@ -0,0 +1,73 @@
+package itrie
+
+import (
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+type Snapshot struct {
+	state *State
+	trie  *Trie
+}
+
+var emptyStateHash = types.StringToHash("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
+
+func (s *Snapshot) GetStorage(addr types.Address, root types.Hash, rawkey types.Hash) types.Hash {
+	var err error
+	var trie *Trie
+
+	if root == emptyStateHash {
+		trie = s.state.newTrie()
+	} else {
+		trie, err = s.state.newTrieAt(root)
+		if err != nil {
+			return types.Hash{}
+		}
+	}
+
+	key := crypto.Keccak256(rawkey.Bytes())
+
+	val, ok := trie.Get(key)
+	if !ok {
+		return types.Hash{}
+	}
+
+	p := &fastrlp.Parser{}
+	v, err := p.Parse(val)
+	if err != nil {
+		return types.Hash{}
+	}
+
+	res := []byte{}
+	if res, err = v.GetBytes(res[:0]); err != nil {
+		return types.Hash{}
+	}
+
+	return types.BytesToHash(res)
+}
+
+func (s *Snapshot) GetAccount(addr types.Address) (*state.Account, error) {
+	key := crypto.Keccak256(addr.Bytes())
+
+	data, ok := s.trie.Get(key)
+	if !ok {
+		return nil, nil
+	}
+	var account state.Account
+	if err := account.UnmarshalRlp(data); err != nil {
+		return nil, err
+	}
+	return &account, nil
+}
+
+func (s *Snapshot) Commit(objs []*state.Object) (state.Snapshot, []byte) {
+	trie, root := s.trie.Commit(objs)
+
+	return &Snapshot{trie: trie, state: s.state}, root
+}
+
+func (s *Snapshot) GetCode(hash types.Hash) ([]byte, bool) {
```

Comment:
Nitpick: 
could you move this function above `Commit`, so that all the functions from `readSnapshot` are grouped together?